pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -214,33 +216,52 @@ func (s *AsyncAgentStore) GetUsers(limit int) []api.User {
 type PollerWorker struct {
 	// AgentCommands is a list of commands executed on the guestAgent
 	AgentCommands []AgentCommand
+
+	// InfoTypes defines the type of guest info to fetch (if applicable)
+	InfoTypes libvirt.DomainGuestInfoTypes
+
 	// CallTick is how often to call this set of commands
 	CallTick time.Duration
 }
 
 type agentCommandsExecutor func(commands []AgentCommand)
 
 // Poll is the call to the guestagent.
-func (p *PollerWorker) Poll(execAgentCommands agentCommandsExecutor, closeChan chan struct{}, initialInterval time.Duration) {
-	log.Log.Infof("Polling command: %v", p.AgentCommands)
+func (p *PollerWorker) Poll(
+	execAgentCommands agentCommandsExecutor,
+	fetchGuestInfo func(infoTypes libvirt.DomainGuestInfoTypes),
+	closeChan chan struct{},
+	initialInterval time.Duration,
+) {
+	genericPoller(func() {
```

Comment:
Why have `genericPoller` when always the same func is passed to it? I'd keep `Poll` and extend it.

Also, in theory it could do both operations, instead of only one of the two.

Reply:
Yep let's remove it. I'd prefer to keep the if condition to execute one of the functions.